### PR TITLE
browserpass: fix dynamic loader

### PR DESCRIPTION
**Pull request by kirillrdy**
###### Description of changes

hopefully fixes
https://github.com/NixOS/nixpkgs/issues/236074

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/browserpass/default.nix [hunk 1]
```diff
@@ -46,7 +46,7 @@ buildGoModule rec {
 
   installPhase = ''
     make install
-
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $out/bin/browserpass
```
Comment (mweinelt) [inline pkgs/tools/security/browserpass/default.nix:49]: Wondering if autopatchElfHook couldn't take care of that.

Comment (kirillrdy) [reply pkgs/tools/security/browserpass/default.nix]: works indeed

**Review by chvp — APPROVED** [review]
Can confirm that this fixes the issue for me (on x86_64-linux).

**Comment by ncfavier** [comment]
[Broke eval on darwin](https://hydra.nixos.org/job/nixpkgs/trunk/browserpass.x86_64-darwin), where `autoPatchelfHook` isn't supported.

**Comment by kirillrdy** [comment]
@ncfavier greetings

thank you for raising this issue. I am a bit confused though, following https://hydra.nixos.org/job/nixpkgs/trunk/browserpass.x86_64-darwin I cant see any errors. 

I also don't understand why autoPatchelfHook would break eval, does it `throw` error ?

**Comment by kirillrdy** [comment]
https://github.com/NixOS/nixpkgs/pull/236749

**Comment by ncfavier** [comment]
You can go to the "Checks" tab of this PR, click "OfBorg" and look at the logs for one of the two failed darwin jobs: https://logs.ofborg.org/?key=nixos/nixpkgs.236258&attempt_id=5cd11477-b9b4-4554-943a-c9b18c1094bc

autoPatchelfHook has `meta.platforms` set to `linux`, so attempting to evaluate it on darwin breaks.

Apparently the darwin equivalent is `fixDarwinDylibNames` (cf https://github.com/NixOS/nixpkgs/pull/163924), do we need to add that as well?

**Comment by kirillrdy** [comment]
@ncfavier I also have https://github.com/NixOS/nixpkgs/pull/236517 which in theory should inform us if we need something like fixDarwinDylibNames

**Comment by ncfavier** [comment]
Good idea. Can you merge the two PRs into one, so that we can test your fix?

**Comment by kirillrdy** [comment]
> Good idea. Can you merge the two PRs into one, so that we can test your fix?

done

**Pull request merged**